packages/client/src/components/project/input/Buttons.tsx

```diff
@@ -48,7 +48,9 @@ export function StartButton(props: { id: string; selected: GetProjectData_projec
 
   const dockerState = useQuery<GetDockerState>(GET_DOCKER_STATE, { pollInterval: 5000 });
   const datasets = useQuery<GetDatasets, GetDatasets>(GET_DATASETS);
-  const datasetNames = props.selected.map((dataset) => dataset.name);
+
+  let datasetName = "";
+  if (props.selected !== null) datasetName = props.selected.name;
```

Comment:
This seems like a weird pattern. Can we move the `props.selected !== null` into the 2 if statements on lines 65 and 75 and remove the datasetName variable?